source/agora/node/main.d

```diff
@@ -81,6 +83,12 @@ private int main (string[] args)
         return 1;
     }
 
-    scope(exit) if (node !is null) node.shutdown();
+    scope(exit){
+        if (node_listener_tuple != NodeListenerTuple.init) with (node_listener_tuple)
+        {
+            node.shutdown();
+            http_listener.stopListening;
+        }
+    }
```

Comment:
```suggestion
    scope (exit)
    {
        if (node_listener_tuple != NodeListenerTuple.init) with (node_listener_tuple)
        {
            node.shutdown();
            http_listener.stopListening;
        }
    }
```

According to our style guide.

P.S. we plan on adding a linter & style check to our CI at some point - in the meantime we have to make these mentions through comments. Sorry for being pedantic.